src/App.vue

```diff
@@ -198,10 +198,8 @@ export default {
     MonacoEditor,
   },
   watch: {
-    mode(mode) {
-      if (mode === 'preview') {
-        this.previewData = this.previewInput ? JSON.parse(this.previewInput) : null;
-      }
+    mode() {
+      this.previewData = this.previewInput ? JSON.parse(this.previewInput) : null;
```

Comment:
Just curious, what was the reason for this change? I don't seem to get any validation errors when setting the data type. For example, I drag a line input, change it's type to `Integer`, and have no errors or issues submitting string data with the form.

Reply:
Some times the data type validation is not updated in preview mode

Steps to reproduce:

1. Add and input with Integer data type
2. Go to preview
3. Put a valid integer (it works), put an invalid integer (and show the validation feedback)
4. Go to design mode
5. Change the Data Type to Decimal
6. Go to preview
7. Put an valid decimal but the validation fails, put an integer passes
8. Finally when you change to design and then back to preview, the validation works as expected

Debugging I see that some times (in step 6)  when we back to preview mode, the v-model for this controls are not properly bind to previewData, keeping it value to undefined even if the user changes the value in the input control.

Then I see that if always setup the `previewData` when the mode changes, this issue doesn't happens.

Reply:
Ok I found the issue, in the vue-form-elements PR, caused by modifying directly the transientData inside a watcher. Now it was fixed.